packages/headless/src/controllers/commerce/core/breadcrumb-manager/headless-core-breadcrumb-manager.ts

```diff
@@ -0,0 +1,219 @@
+  regular: {
+    toggleSelectActionCreator: toggleSelectFacetValue,
+    toggleExcludeActionCreator: toggleExcludeFacetValue,
+  },
+  numericalRange: {
+    toggleSelectActionCreator: toggleSelectNumericFacetValue,
+    toggleExcludeActionCreator: toggleExcludeNumericFacetValue,
+  },
+  dateRange: {
+    toggleSelectActionCreator: toggleSelectDateFacetValue,
+    toggleExcludeActionCreator: toggleExcludeDateFacetValue,
+  },
+  [facetTypeWithoutExcludeAction]: {
+    toggleSelectActionCreator: toggleSelectCategoryFacetValue,
+  },
+};
+
+/**
+ * @internal
+ * Creates a `BreadcrumbManager` controller instance.
+ *
+ * @param engine - The headless commerce engine.
+ * @param options - The `BreadcrumbManager` options used internally.
+ * @returns A `BreadcrumbManager` controller instance.
+ **/
+export function buildCoreBreadcrumbManager(
+  engine: CommerceEngine,
+  options: CoreBreadcrumbManagerOptions
+): BreadcrumbManager {
+  if (!loadCommerceBreadcrumbManagerReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildController(engine);
+  const {dispatch} = engine;
+
+  const createBreadcrumb = (facetId: string) => {
+    const facet = options.facetResponseSelector(engine.state, facetId);
+
+    if (!facet) {
+      return {
+        facetId,
+        field: '',
+        type: 'regular',
+        values: [],
+      };
+    }
+
+    return {
+      facetId,
+      field: facet.field,
+      type: facet.type,
+      values: facet.values
+        .filter((value) => value.state !== 'idle')
+        .map((selection) => ({
+          value: selection,
+          deselect: () => {
+            if (selection.state === 'selected') {
+              // TODO: Do we need to handle freezing values?
+              dispatch(
+                actions[facet.type].toggleSelectActionCreator({
+                  facetId,
+                  selection,
+                })
+              );
+              dispatch(options.fetchResultsActionCreator());
+            } else if (
+              selection.state === 'excluded' &&
+              facet.type !== facetTypeWithoutExcludeAction
+            ) {
+              dispatch(
+                actions[facet.type].toggleExcludeActionCreator!({
+                  facetId,
+                  selection,
+                })
+              );
+              dispatch(options.fetchResultsActionCreator());
+            }
+          },
+        })),
+    };
+  };
+
+  const commerceFacetSelector = createSelector(
+    (state: CommerceEngineState) => state.facetOrder,
+    (facetOrder): BreadcrumbManagerState => {
+      const breadcrumbs = facetOrder.map(createBreadcrumb) ?? [];
+      return {
+        facetBreadcrumbs: breadcrumbs,
+        hasBreadcrumbs: breadcrumbs.length > 0,
+      };
+    }
+  );
+
+  return {
+    ...controller,
+
+    deselectAll: () => {
+      dispatch(deselectAllBreadcrumbs());
+    },
+
+    deselectBreadcrumb(value: DeselectableValue) {
```

Comment:
Should we expose this method at the manager controller level, or should the deselect method be called directly from the breadcrumbs in the controller state?

Reply:
The core breadcrumb controller (the non-commerce one) exposes both a controller method and a breadcrumb value method to deselect.

I'd prefer there be a single way of deselecting a facet. I'll remove the controller method.